core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -519,7 +519,9 @@ private void processData(Map<String, Set<String>> inputStudySampleMap,
                 inputStudySampleMap.keySet().parallelStream().forEach((String _cancerStudyId) -> {
                     try {
                         for(SampleList sampleList:GetSampleLists.getSampleLists(_cancerStudyId)) {
-							if (sampleList.getSampleListCategory().equals(sampleListCategory)) {
+							if (sampleList.getSampleListCategory().equals(sampleListCategory)
+									|| (sampleListCategory.equals(SampleListCategory.ALL_CASES_IN_STUDY)
```

Comment:
Do we really need to check if the category is `ALL_CASES_IN_STUDY` if the sample list's id is `study_id_all`? Because in our case (DFCI internal portal) all the categories for our `study_id_all` lists are `other`.

Reply:
`sampleListCategory` is based in the user selection. we need to select `study_id_all` only if input user select `ALL` case set in query page

Reply:
Ah, ok.